src/lib/ccf/index.ts

```diff
@@ -12,426 +25,451 @@ import * as GCP_EMBODIED from './gcp-embodied.json';
+   */
+  private processInstances(
+    instances: KeyValuePair[],
+    vendor: string,
+    type: string,
+    maxvCPUs: string
+  ) {
+    instances.forEach((instance: KeyValuePair) => {
+      const vCPU = vendor === 'aws' ? 'Instance vCPU' : 'Instance vCPUs';
+      const cpus = parseInt(instance[vCPU], 10);
+      const consumption =
+        vendor === 'aws'
+          ? this.calculateAWSConsumption(instance, cpus)
+          : this.calculateConsumption(instance, vendor, cpus);
+
+      this.computeInstances[vendor][instance[type]] = {
+        name: instance[type],
+        vCPUs: cpus,
+        consumption,
+        maxvCPUs: parseInt(instance[maxvCPUs], 10),
+      } as IComputeInstance;
+    });
+  }
+
+  /**
+   * Retrieves the list of architectures for a given instance based on the INSTANCE_TYPE_COMPUTE_PROCESSOR_MAPPING.
+   */
+  private getInstanceArchitectures(instance: KeyValuePair): string[] {
+    const architectures = INSTANCE_TYPE_COMPUTE_PROCESSOR_MAPPING[
+      instance['Instance type']
+    ] ?? ['Average'];
+
+    return architectures.map((architecture: string) =>
+      this.resolveAwsArchitecture(architecture)
+    );
+  }
+
+  /**
+   * Calculates the average minimum and maximum watts consumption for AWS instances based on the provided architectures.
+   */
+  private calculateAWSAverageWatts(architectures: string[]) {
+    const awsInstance = this.instanceUsage['aws'];
+    let minWatts = 0.0;
+    let maxWatts = 0.0;
+    let count = 0;
+
+    architectures.forEach((architecture: string) => {
+      minWatts += awsInstance[architecture]['Min Watts'];
+      maxWatts += awsInstance[architecture]['Max Watts'];
+      count += 1;
+    });
+
+    return {minWatts: minWatts / count, maxWatts: maxWatts / count};
+  }
+
+  /**
+   * Calculates the consumption metrics (idle, 10%, 50%, 100%, minWatts, maxWatts) for a given compute instance.cv
+   */
+  private calculateConsumption(
+    instance: KeyValuePair,
+    vendor: string,
+    cpus: number
+  ) {
+    const architecture =
+      instance['Microarchitecture'] in this.instanceUsage[vendor]
+        ? instance['Microarchitecture']
+        : 'Average';
+
+    return {
+      idle: 0,
+      tenPercent: 0,
+      fiftyPercent: 0,
+      hundredPercent: 0,
+      minWatts: this.instanceUsage[vendor][architecture]['Min Watts'] * cpus,
+      maxWatts: this.instanceUsage[vendor][architecture]['Max Watts'] * cpus,
+    };
+  }
+
+  /**
+   * Calculates the consumption metrics (idle, 10%, 50%, 100%, minWatts, maxWatts) for a given compute instance.
+   */
+  private calculateAWSConsumption(instance: KeyValuePair, cpus: number) {
+    const architectures = this.getInstanceArchitectures(instance);
+    const {minWatts, maxWatts} = this.calculateAWSAverageWatts(architectures);
+
+    return {
+      idle: parseFloat(instance['Instance @ Idle'].replace(',', '.')),
```

Comment:
you can convert parseFloat, replace to one reusable function and apply for all of props here

Reply:
done